scripts/patchseq_counts.qmd

```diff
@@ -0,0 +1,85 @@
+---
+title: "patchseq"
+---
+
+## TRANSCRIPTOMIC DATA: read counts
+
+```{python}
+import numpy as np
+import pandas as pd
+import scipy
+```
+
+```{python}
+data_exons = pd.read_csv('m1_patchseq_exon_counts.csv.gz',
+                   na_filter=False, index_col=0)
+exonCounts = data_exons.values.transpose()
+exonCounts = scipy.sparse.csr_matrix(exonCounts)
+genes = np.array(data_exons.index)
+```
+
+## exon data introspection
+```{python}
+data_exons.columns # patchseq cell names
+data_exons.index # patchseq genes
+
+```
+
+
+```{python}
+data_introns = pd.read_csv('count_files/m1_patchseq_intron_counts.csv.gz',
+                   na_filter=False, index_col=0)
+intronCounts = data_introns.values.transpose()
+intronCounts = scipy.sparse.csr_matrix(intronCounts)
+```
+
+# TRANSCRIPTOMIC DATA: gene lengths
+
+```{python}
+data = pd.read_csv('count_files/gene_lengths.txt')
+exonLengths = data['exon_bp'].values
+intronLengths = data['intron_bp'].values
+```
+
+
+```{python}
+print('\nBOTH')
+
+seqdepths     = np.array(np.sum(intronCounts+exonCounts,   axis=1))
+genesdetected = np.array(np.sum(intronCounts+exonCounts>0, axis=1))
```

Comment:
Does this double-count the number of genes?  

It seems `genesdetected` might just be informational and not used downstream, so that might not matter, but I thought I'd ask since most genes will presumably have values > 0 for both `intronCounts` and `exonCounts`.

Reply:
The Numpy syntax is a bit misleading if you're not familiar with how the array based math works. So axis=1 indicates that we're working with the arrays of counts column-wise (ie. generating counts per cell). For each cell, where the intronCount plus the exonCount is > 0 (ie. there was gene expression detected for either type) count it as one gene detected. Sum up across all genes to get a count of the number of genes detected for each cell. You end up with an array of the number of genes detected per cell.

Hope that helps.

Reply:
Useful explanation, thanks!